org-code-javabuilder/lib/src/main/java/org/code/javabuilder/UserCodeCompiler.java

```diff
@@ -89,4 +92,107 @@ private CompilationTask getCompilationTask(DiagnosticCollector<JavaFileObject> d
     // create compilation task
     return compiler.getTask(null, fileManager, diagnostics, optionList, null, files);
   }
+
+  /**
+   * For specific compiler errors we want to return custom messages. Check if the given diagnostic
+   * is due to one of those errors and return the more specific message, or null if not.
+   *
+   * @param diagnostic
+   * @return custom compiler error or null
+   */
+  private String getCustomCompilerError(Diagnostic<? extends JavaFileObject> diagnostic) {
+    // Check if the compiler error is due to the student importing java.lang.System
+    // directly and give a more helpful message.
+    if (diagnostic.getCode().equals("compiler.err.already.defined.single.import")
+        && diagnostic.getMessage(Locale.US).contains("org.code.lang.System")) {
+      return "Import of java.lang.System is not supported.";
+    }
+    return null;
+  }
```

Comment:
nit: could pull this logic into `getCompilerError` so you don't need to call two different methods to get the right error message?

Reply:
The reason we have two separate methods is if we get a result from this method we don't show any more compiler errors, but for the generic errors we show all of them.

Reply:
Oh yep, I was just thinking we could early return the custom error message if the diagnostic matches the import error, and if not, return the full message? Maybe I'm misunderstanding though

Reply:
They have slightly different behavior after we get the string. We get a list of Diagnostics from the compiler and go through them one by one in a for loop on line 41. If we see this specific message we don't want to show any more diagnostics in that list, because they will be confusing to students and all we want to show them is this no import message. But for the general case we do want to show every compiler error (for example, they referenced multiple methods incorrectly).

Reply:
Ahhh right sorry I missed the break statement :) thanks for clarifying!